embed/config.go

```diff
@@ -346,14 +331,27 @@ func (cfg Config) ElectionTicks() int { return int(cfg.ElectionMs / cfg.TickMs)
 
 // IsDefaultHost returns the default hostname, if used, and the error, if any,
 // from getting the machine's default host.
-func (cfg Config) IsDefaultHost() (string, error) {
-	if len(cfg.APUrls) == 1 && cfg.APUrls[0].String() == DefaultInitialAdvertisePeerURLs {
-		return defaultHostname, defaultHostStatus
+func (cfg *Config) IsDefaultHost() (string, error) {
+	h, herr := netutil.GetDefaultHost()
+	if len(cfg.APUrls) == 1 {
+		ip, port, err := net.SplitHostPort(cfg.APUrls[0].Host)
+		if err != nil {
+			return "", err
+		}
+		if ip == "0.0.0.0" {
+			cfg.APUrls[0] = url.URL{Scheme: cfg.APUrls[0].Scheme, Host: fmt.Sprintf("%s:%s", h, port)}
```

Comment:
This function is misleading now; usually `Is` functions are idempotent. Here, `cfg.IsDefaultHost()` won't necessarily return the same value on subsequent calls. Probably should just have a separate function do to the advertise IP override.